src/bptl/work_units/xential/views.py

```diff
@@ -0,0 +1,73 @@
+from uuid import UUID
+
+from django.http import HttpResponse, HttpResponseRedirect
+from django.views import View
+
+from rest_framework import status, views
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from bptl.work_units.xential.models import XentialTicket
+from bptl.work_units.xential.serializers import CallbackDataSerializer
+
+from ...tasks.base import check_variable
+from .client import DRC_ALIAS, XENTIAL_ALIAS, get_client
+from .handlers import on_document_created
+from .utils import get_xential_base_url, parse_xml
+
+
+class DocumentCreationCallbackView(views.APIView):
+    permission_classes = []
+
+    def post(self, request: Request) -> Response:
+        # The callback sends the base64 encoded document and the BPTL ticket ID as XML.
+        callback_data = parse_xml(request.data)
+
+        serializer = CallbackDataSerializer(data=callback_data)
+        serializer.is_valid(raise_exception=True)
+
+        bptl_ticket_uuid = serializer.validated_data["bptl_ticket_uuid"]
+
+        # Retrieve the task
+        xential_ticket = XentialTicket.objects.get(bptl_ticket_uuid=bptl_ticket_uuid)
+        task = xential_ticket.task
+
+        # Create the document in the Document API
+        variables = task.get_variables()
+        document_properties = check_variable(variables, "documentMetadata")
+        document_properties["inhoud"] = serializer.validated_data["document"]
+
+        drc_client = get_client(task, DRC_ALIAS)
+        drc_client.post("enkelvoudiginformatieobjecten", json=document_properties)
+
+        # Notify camunda that the document has been created
+        on_document_created(task)
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class InteractiveDocumentView(View):
+    def get(self, request: Request, uuid: UUID) -> HttpResponse:
+        # With the BPTL specific UUID, we can retrieve the Xential ticket ID
+        xential_ticket = XentialTicket.objects.get(bptl_ticket_uuid=uuid)
+
+        xential_client = get_client(xential_ticket.task, XENTIAL_ALIAS)
+
+        # Step 1: Retrieve XSessionID
+        xsession_id_url = "auth/whoami"
+        response_data = xential_client.post(xsession_id_url)
+        headers = {"Cookie": f"XSessionID={response_data['XSessionId']}"}
```

Comment:
I feel this should be part of some `XentialClient` subclass of `JSONClient` to abstract it away in the client itself.